src/api/DeviceInfo.h

```diff
@@ -20,7 +20,7 @@
  */
 + (DeviceInfo *)deviceInfo:(NSString *)memberId keys:(NSArray<Key *> *)keys;
 
-@property (atomic, readonly) NSString *memberId;
-@property (atomic, readonly) NSArray<Key *> *keys;
+@property (nonatomic, readonly) NSString *memberId;
```

Comment:
everything I know about "nonatomic" , I learned in the last couple of minutes by Googling and reading StackOverflow. I'm approving this with a "If Sibin thinks it's a good idea, it's probably a good idea." If you're thinking "Yes, it's a good idea," then great. But if you're thinking "Oh gosh, I don't really know if it's a good idea; I was hoping for an informed second opinion from a knowledgable reviewer," I can go study some more.

Reply:
atomic is designed to synchronize all the write/read methods to the property. e.g. you are going to use this method like a semaphore. But in this case, the only content change is in the object initialization, then the object will never be changed. I don't think we need atomic here. whenever atomic is not mandatory, we shall use nonatomic to improve the performance in objc c.

Reply:
What's more, it is used if this property can be access by multi-thread. but afaik, all we used today is only on main thread. so In the objc sdk, I don't think we shall use atomic anywhere.

Reply:
What's more and more, atomic is only for primitive data types. so in this case, only the pointer is thread-safe but no one can grant the content thread-safe by this atomic. So it is useless here.